core/emulate.c

```diff
@@ -31,24 +31,26 @@
 #include "include/emulate.h"
 
 /* Instruction flags */
-/* Instruction ignores destination original value */
-#define INSN_MOV     ((uint64_t)1 <<  0)
+/* Instruction does not read from destination */
+#define INSN_DST_NR  ((uint64_t)1 <<  0)
```

Comment:
nit: Since it wouldn't make sense to use `DST_NW` and `DST_NR` at the same time, how about allocating 2 bits for this flag (or property)?

```c
enum {
    DST_READWRITE = 0,  /* default */
    DST_WRITEONLY,
    DST_READONLY,
    /* DST_NONE is invalid */
}
```

This is not a required change, but if you are willing to do it, please also create macros for getting and setting this 2-bit flag.

Reply:
My reasoning is that instruction flags are used to signal quirks and other less-common behaviors. Many x86 instructions read and write the destination register (e.g. `add eax, ecx`), consequently the less-common behavior is **not** reading/writing it, otherwise we would be making everything unnecessarily verbose by flagging most instructions with `DST_READWRITE`.

Note that by having "not-read" and "not-write" flags we also save ourselves from having to merge these two under a single "not_readwrite" flag, because no instruction satisfies that property.

Reply:
Sure, that works, thanks for explaining it in detail! I also wanted to avoid flagging most instructions with `DST_READWRITE`, which is why I assigned it the zero value... But I agree that we should be consistent and stick to the one-bit-per-flag design.